portal/views/external_assets.py

```diff
@@ -2,31 +2,79 @@
 
 Primarily LifeRay content access - via exposed APIs for portal side caching
 """
-from flask import Blueprint, current_app
-import requests
+from flask import Blueprint, current_app, request
+from requests.exceptions import ConnectionError
+from ..models.app_text import localize_url, time_request
 
 asset_api = Blueprint('asset', __name__)
 
 
+def get_request(url, params=None):
+    """helper for fetching response data from requested URL
+
+    :param url: the URL to pull content from
+    :param params: optional, if provided, use as
+        parameters to the requested URL
+
+    """
+    error_msg = ''
+    try:
+        response = time_request(url, params).json()
+    except ValueError:  # raised when no json is available in response
+        if response.status_code == 200:
+            current_app.logger.warning(
+                "Request did not return json: {}".format(url))
+            return response.text
+        else:
+            error_msg = (
+                "Could not retrieve remote content - " "{} {}".format(
+                    response.status_code, response.reason))
+    except ConnectionError:
+        error_msg = (
+                    "Could not retrieve remove content - Server could not be "
+                    "reached")
+    if error_msg:
+        current_app.logger.error(error_msg + ": {}".format(url))
```

Comment:
I don't think this Liferay-specific error handling code is strictly necessary. I would say the `ValueError` exception bubbled up as a `500`, would be enough logging (we log all exceptions), but the Liferay URL is essential for debugging.
Maybe just catch the `ValueError`, call `current_app.logger.error()` then `reraise` the caught exception?

Serving valid content is the responsibility of Liferay and I would prefer we don't normalize Liferay's failures to deliver content properly.

Reply:
It is a slippery slope - I think we'd do well to fail noisily when there are problems.
It is very helpful to include any details in the exception text that might save time tracking down the source.